src/plugins/mf_priority.cpp

```diff
@@ -198,6 +198,71 @@ int check_queue_factor (flux_plugin_t *p,
 }
 
 
+/*
+ * Add held job IDs to a JSON array to be added to a bank_info JSON object.
+ */
+static json_t *add_held_jobs (
+                            const std::pair<std::string, struct bank_info> &b)
+{
+    json_t *held_jobs, *jobid = NULL;
+
+    // add any held jobs to a JSON array
+    if (!(held_jobs = json_array ()))
+        goto error;
+
+    for (auto const &j : b.second.held_jobs) {
+        jobid = json_integer (j);
+        if (!jobid)
+            goto error;
+
+        if (json_array_append (held_jobs, jobid) < 0)
+            goto error;
+        
+        json_decref (jobid);
+    }
+
+    return held_jobs;
+error:
+    json_decref (held_jobs);
+    json_decref (jobid);
+    return NULL;
+}
+
+
+/*
+ * Create a JSON object for a bank that a user belongs to.
+ */
+static json_t *pack_bank_info_object (
+                            const std::pair<std::string, struct bank_info> &b)
+{
+    json_t *bank_info, *held_jobs = NULL;
+
+    bank_info = json_object ();
+    if (!bank_info)
+        goto error;
+
+    held_jobs = add_held_jobs (b);
```

Comment:
Need to handle the case where `add_held_jobs()` returns NULL.

Reply:
Yes, thanks for pointing this out - I just squashed and pushed a fix to add a check for `held_jobs` after this function call to check for `NULL`.